connector/connect/client/jvm/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -283,6 +283,16 @@ class Dataset[T] private[sql] (
   def printSchema(level: Int): Unit = println(schema.treeString(level))
   // scalastyle:on println
 
+  /**
+   * Returns an approximate size in bytes of the Dataset.
+   *
+   * @group basic
+   * @since 4.0.0
+   */
+  def sizeInBytesApproximation(): Long = {
```

Comment:
It seems the size may exceed `Long.MaxValue`:

https://github.com/apache/spark/blob/111529e0dea68bf5343cc6aabba53b59e5d21830/core/src/main/scala/org/apache/spark/util/Utils.scala#L1147-L1159

Reply:
Thank you! I missed that. I changed the type to BigInteger (bytes[] in proto)